packages/grid/_modules_/grid/models/colDef/gridColDef.ts

```diff
@@ -162,7 +164,21 @@ export interface GridColDef {
   disableExport?: boolean;
 }
 
-export type GridColumns = GridColDef[];
+export interface GridActionsColDef extends GridColDef {
+  /**
+   * Type allows to merge this object with a default definition [[GridColDef]].
+   * @default 'actions'
+   */
+  type: 'actions';
+  /**
+   * Function that returns the actions to be shown.
+   * @param {GridRowParams} params The params for each row.
+   * @returns {React.ReactElement<GridActionsCellItemProps>[]} An array of [[GridActionsCell]] elements.
+   */
+  getActions: (params: GridRowParams) => React.ReactElement<GridActionsCellItemProps>[];
+}
+
+export type GridColumns = (GridColDef | GridActionsColDef)[];
 export type GridColTypeDef = Omit<GridColDef, 'field'> & { extendType?: GridNativeColTypes };
 
 export interface GridStateColDef extends GridColDef {
```

Comment:
We are loosing the `GridActionsColDef` in `GridStateColDef`

I would rather create a `GridBasicColDef` which equals the current `GridColDef` and have 

```ts
export type GridActionsColDef extends GridBasicColDef { ... }

export type GridColDef = GridBasicColDef | GridActionsColDef`
```

Reply:
> I would rather create a GridBasicColDef which equals the current GridColDef and have

Then we would have to update other [parts](https://material-ui.com/components/data-grid/columns/#column-definitions) of the docs too which is out of the scope of this PR. My initial idea was only to experiment with creating other colDefs types.

> We are loosing the GridActionsColDef in GridStateColDef

If the user adds a type guard it works:

<img width="500" src="https://user-images.githubusercontent.com/42154031/130670517-1e8999a9-1359-42d6-90ce-1a42257e9105.png" alt="image" style="max-width:100%;">

Reply:
Yeah but the typing is kind of wrong here
If you type the param of `hasActions`, Typescript complains because `GridActionsColDef` is not a subset of `GridStateColDef`.

![image](https://user-images.githubusercontent.com/3309670/130746155-46ea1b8a-6591-4e68-8dae-fefaa32d1b94.png)


What I am proposing is to have `GridStateColDef` to be accurate about what is really inside it.

```ts
export interface GridBasicColDef {
  field: string;
  
  // ... all the other fields
}

export type GridActionsColDef extends GridBasicColDef { ... }

export type GridColDef = GridBasicColDef | GridActionsColDef`

export type GridColumns = GridColDef[]

export interface GridStateColDef extends GridColDef {
  computedWidth: number;
}
```

Or if you don't want to touch `GridColDef` 

```ts
export interface GridColDef {
  field: string;
  
  // ... all the other fields
}

export type GridActionsColDef extends GridColDef { ... }

export type GridEnrichedColDef = GridColDef | GridActionsColDef`

export type GridColumns = GridEnrichedColDef[]

export interface GridStateColDef extends GridEnrichedColDef {
  computedWidth: number;
}
```